// protobuf code generated by AST_protobuf.tpl
syntax = "proto3";



package swdv.ifex_scheduler;

// Generic result message
message operation_result {
  bool result = 1;
}

// IFEX Namespace core
// IFEX Enum job_status_t
enum job_status_t {
    PENDING = 0;
    RUNNING = 1;
    COMPLETED = 2;
    FAILED = 3;
    CANCELLED = 4;
}

// IFEX Enum view_type_t
enum view_type_t {
    DAY = 0;
    WEEK = 1;
    MONTH = 2;
    RANGE = 3;
}

// IFEX Struct job_t
message job_t {
  string id = 1;
  string title = 2;
  string service = 3;
  string method = 4;
  string parameters = 5;
  string scheduled_time = 6;
  string recurrence_rule = 7;
  string end_time = 8;
  job_status_t status = 9;
  string created_at = 10;
  string updated_at = 11;
  string executed_at = 12;
  string next_run_time = 13;
  string error_message = 14;
  string result = 15;
  string service_address = 16;
}

// IFEX Struct job_filter_t
message job_filter_t {
  string start_date = 1;
  string end_date = 2;
  string service = 3;
  job_status_t status = 4;
  bool has_status_filter = 5;
  bool include_completed = 6;
}

// IFEX Struct job_create_t
message job_create_t {
  string title = 1;
  string service = 2;
  string method = 3;
  string parameters = 4;
  string scheduled_time = 5;
  string recurrence_rule = 6;
  string end_time = 7;
  string service_address = 8;
}

// IFEX Struct job_update_t
message job_update_t {
  string title = 1;
  string scheduled_time = 2;
  string recurrence_rule = 3;
  string end_time = 4;
  string parameters = 5;
  string service_address = 6;
}

// IFEX Method create_job
message create_job_request {
  job_create_t job = 1;
}

message create_job_response {
  bool success = 1;
  string job_id = 2;
  string message = 3;
}

service create_job_service {
  rpc create_job(create_job_request) returns (create_job_response);
}

// IFEX Method get_jobs
message get_jobs_request {
  job_filter_t filter = 1;
}

message get_jobs_response {
  bool success = 1;
  repeated job_t jobs = 2;
}

service get_jobs_service {
  rpc get_jobs(get_jobs_request) returns (get_jobs_response);
}

// IFEX Method get_job
message get_job_request {
  string job_id = 1;
}

message get_job_response {
  bool success = 1;
  job_t job = 2;
  string message = 3;
}

service get_job_service {
  rpc get_job(get_job_request) returns (get_job_response);
}

// IFEX Method update_job
message update_job_request {
  string job_id = 1;
  job_update_t updates = 2;
}

message update_job_response {
  bool success = 1;
  string message = 2;
}

service update_job_service {
  rpc update_job(update_job_request) returns (update_job_response);
}

// IFEX Method delete_job
message delete_job_request {
  string job_id = 1;
}

message delete_job_response {
  bool success = 1;
  string message = 2;
}

service delete_job_service {
  rpc delete_job(delete_job_request) returns (delete_job_response);
}

// IFEX Method get_calendar_view
message get_calendar_view_request {
  view_type_t view_type = 1;
  string date = 2;
  string timezone = 3;
}

message get_calendar_view_response {
  bool success = 1;
  string start_date = 2;
  string end_date = 3;
  repeated job_t jobs = 4;
}

service get_calendar_view_service {
  rpc get_calendar_view(get_calendar_view_request) returns (get_calendar_view_response);
}





